backends-common/elasticsearch/src/main/java/org/apache/james/backends/es/ClientProvider.java

```diff
@@ -18,36 +18,110 @@
  ****************************************************************/
 package org.apache.james.backends.es;
 
+import java.io.File;
 import java.io.IOException;
+import java.security.KeyManagementException;
+import java.security.KeyStoreException;
+import java.security.NoSuchAlgorithmException;
 import java.time.Duration;
 import java.time.LocalDateTime;
 
 import javax.annotation.PreDestroy;
 import javax.inject.Inject;
 import javax.inject.Provider;
+import javax.net.ssl.SSLContext;
 
 import org.apache.commons.lang3.time.DurationFormatUtils;
 import org.apache.http.HttpHost;
+import org.apache.http.auth.AuthScope;
+import org.apache.http.auth.UsernamePasswordCredentials;
+import org.apache.http.client.CredentialsProvider;
+import org.apache.http.conn.ssl.NoopHostnameVerifier;
+import org.apache.http.impl.client.BasicCredentialsProvider;
+import org.apache.http.impl.nio.client.HttpAsyncClientBuilder;
+import org.apache.http.ssl.SSLContextBuilder;
+import org.apache.http.ssl.TrustStrategy;
+import org.apache.james.backends.es.ElasticSearchConfiguration.HostScheme;
+import org.apache.james.backends.es.ElasticSearchConfiguration.SSLTrustStore;
 import org.elasticsearch.client.RestClient;
 import org.elasticsearch.client.RestHighLevelClient;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.github.fge.lambdas.Throwing;
 import com.google.common.annotations.VisibleForTesting;
+
 import reactor.core.publisher.Mono;
 import reactor.core.scheduler.Schedulers;
 
 public class ClientProvider implements Provider<RestHighLevelClient> {
 
+    private static class HttpAsyncClientConfigurer {
+
+        private static final TrustStrategy TRUST_ALL = (x509Certificates, authType) -> true;
+
+        private final ElasticSearchConfiguration configuration;
+
+        private HttpAsyncClientConfigurer(ElasticSearchConfiguration configuration) {
+            this.configuration = configuration;
+        }
+
+        private HttpAsyncClientBuilder configure(HttpAsyncClientBuilder builder) {
+            configureAuthentication(builder);
+            configureSSL(builder);
+
+            return builder;
+        }
+
+        private void configureSSL(HttpAsyncClientBuilder builder) {
+            if (configuration.getHostScheme().equals(HostScheme.HTTP)) {
+                return;
+            }
+
+            try {
```

Comment:
We should fail here if the scheme is not HTTPS (avoids errors when adding some new HostScheme in the future...)